hazelcast/src/main/java/com/hazelcast/map/impl/query/QueryEngineImpl.java

```diff
@@ -125,15 +125,16 @@ public Result execute(Query query, Target target) {
 
     private Query adjustQuery(Query query) {
         IterationType retrievalIterationType = getRetrievalIterationType(query.getPredicate(), query.getIterationType());
-        Query adjustedQuery = Query.of(query).iterationType(retrievalIterationType).build();
-        if (adjustedQuery.getPredicate() instanceof PagingPredicateImpl) {
-            ((PagingPredicateImpl) adjustedQuery.getPredicate()).setIterationType(query.getIterationType());
-        } else {
-            if (adjustedQuery.getPredicate() == Predicates.alwaysTrue()) {
-                queryResultSizeLimiter.precheckMaxResultLimitOnLocalPartitions(adjustedQuery.getMapName());
-            }
+        Query.QueryBuilder builder = Query.of(query).iterationType(retrievalIterationType);
+        if (query.getPredicate() instanceof PagingPredicateImpl) {
```

Comment:
What about `MultiPartitionPredicateImpl`?

Reply:
I think in this case we will be ok as this is `PagingPredicateImpl` we are checking for.

From what I can tell, this wraps the inner predicate so when we clone this we only clone the `PagingPredicateImpl` to prevent the iteration type from being clobbered. We don't clone the inner predicate which would require the check for `Multi/PartitionPredicateImpl`